### PR TITLE
W-13885636 | Adding support for Java 17 - check TODOs

### PR DESCRIPTION
**Pull request by marianborzik**
TODO:

remove snapshot postfix from SOAP Engine and Validation modules once, they are released

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,14 +23,15 @@
...
+        <mule.http.connector.version>1.9.0</mule.http.connector.version>
+        <muleSdkApiVersion>0.7.0</muleSdkApiVersion>
         <soap.engine.version>2.0.0-SNAPSHOT</soap.engine.version>
-        <wsdl.parser.version>1.5.0-SNAPSHOT</wsdl.parser.version>
+        <wsdl.parser.version>1.4.7</wsdl.parser.version>
         <activation.version>1.1.1</activation.version>
         <javax.jaxws-api.version>2.3.1</javax.jaxws-api.version>
         <javax.jws-api.version>1.1</javax.jws-api.version>
         <formatterConfigPath>formatter.xml</formatterConfigPath>
+<!--        <jacoco.version>0.8.10</jacoco.version>-->
```
Comment (sabecasismulesoft) [inline pom.xml:35]: Why is the jacoco version commented?

File: pom.xml [hunk 2]
```diff
@@ -23,14 +23,15 @@
     <properties>
         <commons.lang.version>3.12.0</commons.lang.version>
 
-        <mule.http.connector.version>1.7.3</mule.http.connector.version>
-        <muleSdkApiVersion>0.3.0</muleSdkApiVersion>
+        <mule.http.connector.version>1.9.0</mule.http.connector.version>
+        <muleSdkApiVersion>0.7.0</muleSdkApiVersion>
         <soap.engine.version>2.0.0-SNAPSHOT</soap.engine.version>
-        <wsdl.parser.version>1.5.0-SNAPSHOT</wsdl.parser.version>
+        <wsdl.parser.version>1.4.7</wsdl.parser.version>
```
Comment (sabecasismulesoft) [inline pom.xml:29]: Is this version of mule-wsdl-parser java 17 compatible?

**Review by sabecasismulesoft — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (fburman-mule) [reply pom.xml:35]: Uncommented

File: src/main/java/org/mule/extension/ws/internal/WebServiceConsumer.java [hunk 3]
```diff
@@ -19,6 +22,8 @@
 import org.mule.runtime.extension.api.annotation.connectivity.ConnectionProviders;
 import org.mule.runtime.extension.api.annotation.dsl.xml.Xml;
 import org.mule.runtime.extension.api.annotation.error.ErrorTypes;
+import org.mule.sdk.api.annotation.JavaVersionSupport;
+//import org.mule.sdk.api.annotation.JavaVersionSupport;
```
[the remark this answers is not in the document]
Comment (fburman-mule) [reply src/main/java/org/mule/extension/ws/internal/WebServiceConsumer.java:26]: removed

File: src/test/resources/log4j2-test.xml [hunk 4]
```diff
@@ -11,5 +11,3 @@
             <AppenderRef ref="Console"/>
         </AsyncRoot>
     </Loggers>
-
```
Comment (sabecasismulesoft) [inline src/test/resources/log4j2-test.xml:14]: The configuration closing tag is missing. Please add it back

**Review by sabecasismulesoft — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (tomastoloza) [reply pom.xml:29]: No, it's not, it should be 1.4.8-SNAPSHOT for this SNAPSHOT release
```suggestion
        <wsdl.parser.version>1.4.8-SNAPSHOT</wsdl.parser.version>
```

File: src/test/resources/log4j2-test.xml (hunk 4 above)
Comment (fburman-mule) [reply src/test/resources/log4j2-test.xml:14]: done!

File: pom.xml (hunk 2 above)
Comment (fburman-mule) [reply pom.xml:29]: We donn't know, there's still no wsdl parser ready. As this is a snapshot it's not a blocker

File: pom.xml [hunk 5]
```diff
@@ -45,12 +46,12 @@
         <munit.input.directory>src/test/munit</munit.input.directory>
         <munit.output.directory>${basedir}/target/test-mule/munit</munit.output.directory>
         <mavenResourcesVersion>3.3.1</mavenResourcesVersion>
-        <munit.version>2.3.18</munit.version>
-        <mtf.tools.version>1.1.2</mtf.tools.version>
-        <munit.extensions.maven.plugin.version>1.1.2</munit.extensions.maven.plugin.version>
+        <munit.version>3.1.0-rc3</munit.version>
+        <mtf.tools.version>1.2.0-rc2</mtf.tools.version>
```
Comment (tomastoloza) [inline pom.xml:50]: build is failing because there's no rc2
```suggestion
        <mtf.tools.version>1.2.0-rc1</mtf.tools.version>
```

**Review by sabecasismulesoft — APPROVED** [review]

**Pull request merged**